comp/remote-config/rcclient/rcclientimpl/rcclient.go

```diff
@@ -291,14 +294,26 @@ func (rc rcClient) agentConfigUpdateCallback(updates map[string]state.RawConfig,
 
 		// Need to update the log level even if the level stays the same because we need to update the source
 		// Might be possible to add a check in deeper functions to avoid unnecessary work
-		err = rc.settingsComponent.SetRuntimeSetting("log_level", mergedConfig.LogLevel, model.SourceRC)
+		if err := rc.settingsComponent.SetRuntimeSetting("log_level", mergedConfig.LogLevel, model.SourceRC); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	streamLogsSource := config.Datadog().GetSource("logs_config.streaming.enable_streamlogs")
+	if mergedConfig.EnableStreamLogs {
+		if err := rc.settingsComponent.SetRuntimeSetting("enable_streamlogs", mergedConfig.EnableStreamLogs, model.SourceRC); err != nil {
+			errs = append(errs, err)
+		}
+	} else if !mergedConfig.EnableStreamLogs && streamLogsSource == model.SourceRC {
+		config.Datadog().UnsetForSource("logs_config.streaming.enable_streamlogs", model.SourceRC)
 	}
 
 	// Apply the new status to all configs
 	for cfgPath := range updates {
-		if err == nil {
+		if errs == nil {
 			applyStateCallback(cfgPath, state.ApplyStatus{State: state.ApplyStateAcknowledged})
 		} else {
+			err := fmt.Errorf("error while applying remote config: %s", errs)
```

Comment:
Would it make sense to use `github.com/hashicorp/go-multierror` here? We already use it for similar cases elsewhere.

Reply:
I didn't know we use this. Ill adjust my code to that instead 👍🏼